pipeline/splicing/calcVarPriors.py

```diff
@@ -1020,70 +1190,111 @@ def compareRefAltExonLengths(refLength, altLength):
-        deNovoSplicePos = getNewSplicePosition(variant["Pos"], varStrand, slidingWindowInfo["varWindowPosition"],
-                                               slidingWindowInfo["inExonicPortion"], exonicPortionSize)
         if varStrand == "+":
             distanceBetween = wildTypeSplicePos - deNovoSplicePos
         else:
+            wildTypeSplicePos = refExonBounds[varExonNum]["exonEnd"]
             # +1 because of RefSeq numbering
             distanceBetween = deNovoSplicePos - (wildTypeSplicePos + 1)
-        if distanceBetween % 3 == 0:
-            return True
+    else:
+        wildTypeSplicePos = refExonBounds[varExonNum]["exonStart"]
+        if varStrand == "+":
+            distanceBetween = abs(deNovoSplicePos - wildTypeSplicePos)
         else:
-            return False
+            # +1 because of RefSeq numbering
+            distanceBetween = abs((wildTypeSplicePos + 1) - deNovoSplicePos)
+            
+    if distanceBetween % 3 == 0:
+        return True
+    return False
         
 def getPriorProbSpliceRescueNonsenseSNS(variant, boundaries, deNovoDonorInRefAcc=False):
     '''
     Given a variant, determines if there is a possibility of splice rescue
     deNovoDonorInRefAcc argument = True  if looking for deNovoDonor in ref acceptor site, False otherwise
     If there is a possibility of splice rescue, flags variant for further analysis
     Else assigns prior probability of pathogenecity and predicted qualitative ENIGMA class
-    '''
-    varCons = getVarConsequences(variant)
-    # check that variant causes a premature stop codon in an exon
-    if varCons == "stop_gained" and varInExon(variant) == True:
+    Extra flags are included to provide more information about why splicing rescue does or does not occur
+      frameshiftFlag: equals 1 if variant causes a frameshift, 0 if not
+      inExonicPortionFlag: equals 1 if variant IS in exonic portion of highest scoring window, 0 if not
+      CIDomainRegionFlag: equals 1 if truncating region (between new donor and next reference splice acceptor) 
+         includes a clinicall import domain, 0 if not
+      isDivisibleFlag: equals 1 if distance between de novo donor and wild-type donor splice site is NOT divisible by 3, 
+         0 if not
+    '''
+    if varInIneligibleDeNovoExon(variant, donor=True) == True:
+        return {"priorProb": 0.99,
+                "enigmaClass": "class_5",
+                "spliceRescue": 0,
+                "spliceFlag": 0,
+                "frameshiftFlag": "N/A",
+                "inExonicPortionFlag": "N/A",
+                "CIDomainInRegionFlag": "N/A",
+                "isDivisibleFlag": "N/A",
+                "lowMESFlag": "N/A"}
+    # checks that variant causes a premature stop codon in an exon
+    if getVarConsequences(variant) == "stop_gained" and varInExon(variant):
         spliceFlag = 0
         spliceRescue = 0
-        frameshift = 0
+        frameshiftFlag = "-"
```

Comment:
i understand that values are 0 if false and 1 if true, but they're also sometimes "n/a" and sometimes "-", could you explain why all these values are necessary?

Reply:
It only assigns a 0 or 1 if it actually checks for that specific thing. If it doesn't check at all then "-" is assigned and if it checks but it is not relevant it assigns "N/A"

Reply:
perfect! could you document this somewhere?